packages/react/src/SubdomainNavBar/SubdomainNavBar.stories.tsx

```diff
@@ -547,6 +549,18 @@ MobileMenuOpen.play = async ({canvasElement}) => {
   await userEvent.click(canvas.getByLabelText('Menu'))
 }
 
+export const MobileMenuOpenManyItems = Template.bind({})
```

Comment:
The overflowing buttons regression seems to be cropping out the close button too on this story:

https://primer-cabd91a764-26139705.drafts.github.io/storybook/?path=/story/components-subdomainnavbar--mobile-menu-open-many-items

![Screenshot 2023-10-10 at 15 23 37](https://github.com/primer/brand/assets/13340707/f290f49b-63d6-4177-822d-8573fc1f805f)

It doesn't seem to be happening on this story however: 
https://primer-cabd91a764-26139705.drafts.github.io/storybook/?path=/story/components-subdomainnavbar--mobile-menu-open

Possible to take a look into what's happening here please? We can't have the close button being partially out off canvas.